pkg/liqoctl/status/peer/peerinfo.go

```diff
@@ -0,0 +1,352 @@
+	if err != nil {
+		pic.addCollectionError("ForeignClusters", "", err)
+		return
+	}
+
+	remoteClusterNamesNotFound := pic.checkPeerExistence(foreignClusterList)
+
+	foreignClusterListSelected := pic.getForeignClusterListSelected(foreignClusterList)
+
+	for i := range foreignClusterListSelected.Items {
+		fc := &foreignClusterListSelected.Items[i]
+		remoteClusterName := fc.Spec.ClusterIdentity.ClusterName
+		if slices.Contains(remoteClusterNamesNotFound, remoteClusterName) {
+			pic.peerInfoSection.AddSectionWithDetail(remoteClusterName, PeerNotFoundMsg)
+			continue
+		}
+
+		remoteClusterID := fc.Spec.ClusterIdentity.ClusterID
+
+		clusterSection := pic.peerInfoSection.AddSectionWithDetail(remoteClusterName, remoteClusterID)
+
+		pic.addPeerSection(clusterSection, fc)
+
+		pic.addAuthSection(clusterSection, fc)
+
+		err = pic.addNetworkSection(ctx, clusterSection, fc, localClusterName)
+		if err != nil {
+			pic.addCollectionError("Network", "unable to collect Network", err)
+		}
+
+		err = pic.addResourceSection(ctx, clusterSection, remoteClusterID, localClusterName, remoteClusterName)
+		if err != nil {
+			pic.addCollectionError("Resources", "unable to collect Resources", err)
+		}
+	}
+}
+
+// getForeignClusterList returns the list of ForeignCluster objects.
+func (pic *PeerInfoChecker) getForeignClusterListSelected(
+	foreignClusterList *discoveryv1alpha1.ForeignClusterList) *discoveryv1alpha1.ForeignClusterList {
+	foreignClusterListSelected := discoveryv1alpha1.ForeignClusterList{}
+	if len(pic.pio.RemoteClusterNames) == 0 {
+		foreignClusterListSelected = *foreignClusterList
+	} else {
+		for _, rcn := range pic.pio.RemoteClusterNames {
+			fc := getForeignClusterByName(foreignClusterList, rcn)
+			if fc != nil {
+				foreignClusterListSelected.Items = append(foreignClusterListSelected.Items, *fc)
+			} else {
+				foreignClusterListSelected.Items = append(foreignClusterListSelected.Items, discoveryv1alpha1.ForeignCluster{
+					Spec: discoveryv1alpha1.ForeignClusterSpec{
+						ClusterIdentity: discoveryv1alpha1.ClusterIdentity{
+							ClusterName: rcn,
+						},
+					},
+				})
+			}
+		}
+	}
+	return &foreignClusterListSelected
+}
+
+// getForeignClusterByName returns the ForeignCluster with the given name from a ForeignClusterList.
+func getForeignClusterByName(fcList *discoveryv1alpha1.ForeignClusterList, fcName string) *discoveryv1alpha1.ForeignCluster {
+	for i := range fcList.Items {
+		if fcList.Items[i].Spec.ClusterIdentity.ClusterName == fcName {
+			return &fcList.Items[i]
+		}
+	}
+	return nil
+}
+
+// checkPeerExistence checks if the given foreign cluster exists.
+// returns a slice of strings containing the names of the missing peers and a boolean indicating if all the given peers exists.
+func (pic *PeerInfoChecker) checkPeerExistence(fcList *discoveryv1alpha1.ForeignClusterList) []string {
+	var fcNames []string
```

Comment:
It is better to define this as set (i.e., map[string]struct{}) for direct access.